tests/suites/helpers.function

```diff
@@ -107,15 +108,52 @@ typedef enum
  *
  * \param   TEST    The test expression to be tested.
  */
-#define TEST_ASSERT( TEST )                                 \
-    do {                                                    \
-       if( ! (TEST) )                                       \
-       {                                                    \
-          test_fail( #TEST, __LINE__, __FILE__ );           \
-          goto exit;                                        \
-       }                                                    \
+#define TEST_ASSERT( TEST )                                                   \
+    do {                                                                      \
+        if( ! (TEST) )                                                        \
+        {                                                                     \
+            set_test_result( #TEST, __LINE__, __FILE__, TEST_RESULT_FAILED ); \
+            goto exit;                                                        \
+        }                                                                     \
+    } while( 0 )
+
+/**
+ * \brief   This macro tests the expression passed to it as a test step or
+ *          individual test in a test case, comparing result to an expected
+ *          result given as parameter, and skips the test in case the underlying
+ *          alternative implementation doesn't support the feature.
+ *
+ *          It allows a library function to return a value and return an error
+ *          code that can be tested.
+ *          It receives an expected return code and:
+ *          - If the underlying alternative implementation does not support the feature,
+ *          by returning MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED, the test is skipped.
+ *          - If and fails in case the expression
```

Comment:
Grammar is mistake and I'm not even sure what you were trying to say.

Reply:
Actually it's a rebase error, I'll fix